src/graphics/start.c

```diff
@@ -0,0 +1,288 @@
+#include <stdlib.h>
+#include "../../include/raylib.h"
+#include "../../include/raymath.h"
+
+#include "tmp_structs.h"
+#include "start.h"
+#include "constants.h"
+
+void initStartMenu(Menu *menu, int screen_width, int screen_height){
+    menu->menu_button_array= (MenuButton*)malloc(sizeof(MenuButton)*NUM_MENU_BUTTONS);
+    menu->title = TITLE;
+    menu->menu_button_array[0].text=START_BUTTON_TITLE;
+    menu->menu_button_array[1].text=SETTINGS_BUTTON_TITLE;
+    menu->menu_button_array[2].text=STATS_BUTTON_TITLE;
+    menu->menu_button_array[3].text=QUIT_BUTTON_TITLE;
+    for(int i=0; i<NUM_MENU_BUTTONS;i++){
+        menu->menu_button_array[i].button_width =((float)screen_width/5);
+        menu->menu_button_array[i].button_height = ((float)screen_height/10);
+    }
+    menu->button_hover=0;
+    setMenuBounds(menu,screen_width,screen_height);
+}
+
+void initStartGameSettings(StartSettings *start_settings, int screen_width, int screen_height){
+    start_settings->menu_button_array= (MenuButton*)malloc(sizeof(MenuButton)*NUM_SETTINGS_BUTTONS);
+    start_settings->menu_button_array[0].text=BACK_BUTTON_TITLE;
+    start_settings->menu_button_array[1].text=START_BUTTON_TITLE;
+    for(int i=0; i<NUM_SETTINGS_BUTTONS;i++){
+        start_settings->menu_button_array[i].button_width =((float)screen_width/5);
+        start_settings->menu_button_array[i].button_height = ((float)screen_height/10);
+    }
+    start_settings->button_hover=0;
+    setStartGameBounds(start_settings,screen_width,screen_height);
+}
+
+void initSettings(Settings *settings, int screen_width, int screen_height){
+    settings->menu_button_array= (MenuButton*)malloc(sizeof(MenuButton)*NUM_SETTINGS_BUTTONS);
+    settings->menu_button_array[0].text=BACK_BUTTON_TITLE;
+    settings->menu_button_array[1].text="APPLY";
+    for(int i=0; i<NUM_SETTINGS_BUTTONS;i++){
+        settings->menu_button_array[i].button_width =((float)screen_width/5);
+        settings->menu_button_array[i].button_height = ((float)screen_height/10);
+    }
+    settings->button_hover=0;
+    setSettingsBounds(settings,screen_width,screen_height);
+}
+
+void initStatistics(Statistics *statistics, int screen_width, int screen_height){
+    statistics->menu_button_array= (MenuButton*)malloc(sizeof(MenuButton));
+    statistics->menu_button_array[0].text=BACK_BUTTON_TITLE;
+    statistics->menu_button_array[0].button_width =((float)screen_width/5);
+    statistics->menu_button_array[0].button_height = ((float)screen_height/10);
+    statistics->button_hover=0;
+    setStatsBounds(statistics,screen_width,screen_height);
+}
+
+
+void setMenuBounds(Menu *menu,double width,double height){
+    Rectangle start_bounds = {(width/2.0f-menu->menu_button_array[0].button_width/2.0f),height/4.0f,menu->menu_button_array[0].button_width,
```

Comment:
The literals you've used in the calculations of the bound - what are these? can we not exist them to a variable? Are the bounds independent of the width and height of the screen?

Reply:
We discussed this and I hope i made some of it more clear